tensorflow/core/user_ops/chol_grad.cc

```diff
@@ -0,0 +1,115 @@
+//Copyright (c) 2016, Alexander G. de G. Matthews and James Hensman. All rights reserved.
+
+//Based on code for blocked Cholesky reference mode differentiation by Iain Murray.
+//For an explanation see "Differentiation of the Cholesky algorithm" by Iain Murray http://arxiv.org/abs/1602.07527.
+
+#include "tensorflow/core/framework/op.h"
+#include "third_party/eigen3/Eigen/Core"
+
+REGISTER_OP("CholeskyGrad").Input("l: T").Input("l_bar: T").Output("a_bar: T").Attr( "T: {float, double}").Doc("Cholesky backpropagation where l is output of Cholesky algorithm and f is gradient of some loss wrt l");
+
+#include "tensorflow/core/framework/op_kernel.h"
+
+#include "third_party/eigen3/unsupported/Eigen/CXX11/Tensor"
+#include "tensorflow/core/framework/tensor_types.h"
+#include "tensorflow/core/framework/types.h"
+
+using namespace tensorflow;
+
+template <typename T>
+class CholeskyGrad : public OpKernel {
+    public:
+    
+    explicit CholeskyGrad(OpKernelConstruction* context) : OpKernel(context) {}
+
+    using Matrix = Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>;
+    using ConstMatrixMap = Eigen::Map<const Matrix>;
+    using MatrixMap = Eigen::Map<Matrix>;
+    using ConstRef = Eigen::Ref<const Matrix>;
+    using Ref = Eigen::Ref<Matrix>;
+    using lcl_size_t = int;
+
+    void Compute(OpKernelContext* context) override {
+
+    const Tensor& input_tensor_l = context->input(0); ;
```

Comment:
maybe name these input_matrix_l and input_matrix_grad, so it is clearer without having to know the notation in the paper.